client/src/featureform/providers/online_store.py

```diff
@@ -0,0 +1,28 @@
+class OnlineStore:
+    def create_table(self, name, variant, entity_type):
+        raise NotImplementedError()
+
+    def get_table(self, name, variant):
+        raise NotImplementedError()
+
+    def table_exists(self, name, variant):
+        raise NotImplementedError()
+
+    def is_vector_store(self):
```

Comment:
why do we need this?

It's like having an implementation of List, then having an implementation of LinkedList then inherits from List, and then adding a function on the List interface that says `isLinkedList`.

Reply:
We'll need some way of having the equivalent Golang type assertion. Would you suggest/prefer something like this?

```python
from abc import ABC

class OnlineStore(ABC):
    pass

class VectorStore(OnlineStore, ABC):
    pass

class Pinecone(VectorStore)
    pass

p = Pinecone()

if isinstance(store_instance, VectorStore):
    # do something vector specific like create an index
```

Reply:
yeah, just take advantage of the abstraction. Could also have a static method or a function that does it: `isVectorStore(store)`.Making it part of the interface is too much boilerplate.